omod-2.0/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs2_0/ConceptAttributeTypeController2_0Test.java

```diff
@@ -0,0 +1,76 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs2_0;
+
+import org.apache.commons.beanutils.PropertyUtils;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.api.ConceptService;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.web.v1_0.RestTestConstants2_0;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.MainResourceControllerTest;
+import org.springframework.mock.web.MockHttpServletRequest;
+import org.springframework.web.bind.annotation.RequestMethod;
+
+/**
+ * Tests functionality of {@link ConceptAttributeTypeController}.
+ */
+public class ConceptAttributeTypeController2_0Test extends MainResourceControllerTest {
+
+    private ConceptService service;
+
+    /**
+     * @see MainResourceControllerTest#getURI()
+     */
+    @Override
+    public String getURI() {
+        return "conceptattributetype";
+    }
+
+    /**
+     * @see MainResourceControllerTest#getUuid()
+     */
+    @Override
+    public String getUuid() {
+        return RestTestConstants2_0.CONCEPT_ATTRIBUTE_TYPE_UUID;
+    }
+
+    /**
+     * @see MainResourceControllerTest#getAllCount()
+     */
+    @Override
+    public long getAllCount() {
+        return service.getAllConceptAttributeTypes().size();
+    }
+
+    @Before
+    public void before() throws Exception {
+        executeDataSet(RestTestConstants2_0.CONCEPT_ATTRIBUTE_DATA_SET);
```

Comment:
Do you need to execute the above data set?

Reply:
yes @dkayiwa when am testing already the dataset includes a number of `ConceptAttributeTypes` .....so when am testing I test if that number increases by 1

Reply:
Would the test fail if you remove it?

Reply:
no ...let me update it @dkayiwa

Reply:
@dkayiwa when I remove it the getUuid() fails with `org.openmrs.module.webservices.rest.web.response.ObjectNotFoundException`

![Screenshot (57)](https://user-images.githubusercontent.com/86114366/141766474-4c09c774-e231-412e-aba6-508cc8673102.png)